packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts

```diff
@@ -37,11 +46,8 @@ export class QueryDataViewService {
     QueryDataViewService.extensionPath = context.extensionPath;
   }
 
-  private updateWebviewWith(
-    webview: vscode.Webview,
-    queryData: QueryResult<JsonMap>
-  ) {
-    webview.postMessage({
+  private updateWebviewWith(queryData: QueryResult<JsonMap>) {
+    this.currentPanel!.webview.postMessage({
```

Comment:
is `!` necessary here?

Reply:
Yeah, just because of the way the webview lifecycle works, we can't instantiate a webview with out displaying it. So the current panel is assigned with `createOrShowWebview()` [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/8e4b0955b7cc6bb108baa6e9bae0fbb8006c0cd0/packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts#L58)

Instead of when the new dataView class is created

Reply:
Got it, but do you want to change it to optional chaining to avoid NPEs?
```suggestion
    this.currentPanel?.webview.postMessage({
```

Reply:
yeah sounds good to me.